File: spark-scala-maven/src/main/scala/adj/spark/df/GenericSave.scala
package adj.spark.df

import org.apache.spark.sql.SQLContext
import org.apache.spark.SparkConf
import org.apache.spark.api.java.JavaSparkContext
import org.apache.spark.sql.SaveMode

object GenericSave {
  def main(args: Array[String]): Unit = {
    val conf = new SparkConf()
			     			.setAppName(GenericSave.getClass.getSimpleName())
			     			.setMaster("local")
		
		val context    = new JavaSparkContext(conf)
		val sqlContext = new SQLContext(context)
		
		val dataFrame = sqlContext.read.json("hdfs://master:9000/user/spark/data/students.json")
		
		// 指定保存文件的格式
		// dataFrame.select("id", "name", "age").write.format("parquet").save("hdfs://master:9000/user/spark/data/df/student.parquet")
		
		/**
		 * mode
		 * 1.SaveMode.ErrorIfExists
		 * 2.SaveMode.Append
		 * 3.SaveMode.Overwrite
		 * 4.SaveMode.Ignore
		 */
		dataFrame.select("id", "name", "age").write.mode(SaveMode.Overwrite).save("hdfs://master:9000/user/spark/data/df/student.parquet")
  }
}

File: spark-java/src/main/java/adj/spark/df/GenericSave.java
package adj.spark.df;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.sql.DataFrame;
import org.apache.spark.sql.SQLContext;

public class GenericSave {
	public static void main(String[] args) {
		SparkConf conf = new SparkConf()
			     			.setAppName(GenericSave.class.getSimpleName())
			     			.setMaster("local");
		
		JavaSparkContext context    = new JavaSparkContext(conf);
		SQLContext       sqlContext = new SQLContext(context);
		
		DataFrame dataFrame = sqlContext.read().json("hdfs://master:9000/user/spark/data/students.json");
		dataFrame.select("id", "name", "age").write().save("hdfs://master:9000/user/spark/data/df/student.parquet");
		
		dataFrame.show();
		dataFrame.printSchema();
	}
}
